tds.go

```diff
@@ -796,8 +796,11 @@ initiate_connection:
 	}
 
 	// This will panic will never occur unless our code is wrong
-	ch := ctx.Value("fields").(chan map[uint8][]byte)
-	ch <- fields
+	ch := ctx.Value("fields")
+
+	if ch != nil {
+		ch.(chan map[uint8][]byte) <- fields
```

Comment:
A somewhat nitty comment but worth making:

We're mixing two different actions into a single statement here.  For clarity, better to do the cast on one line, and then the "putting fields into the channel" on the next line.  This gives us better readability, and also, if either line fails, the failure cause will be clear from the stack trace line.

Reply:
Like so?
```
	// This will panic will never occur unless our code is wrong
	ch := ctx.Value("fields")

	if ch != nil {
		ch := ch.(chan map[uint8][]byte)
		ch <- fields
	}
```